pydra/engine/tests/test_shelltask_inputspec.py

```diff
@@ -2073,3 +2073,412 @@ def test_shell_cmd_inputs_di(tmpdir, use_validator):
+        fields=[
+            (
+                "out_1",
+                attr.ib(
+                    type=File,
+                    metadata={
+                        "help_string": "fictional output #1",
+                        "output_file_template": "{out_dir}/test_1.nii",
+                        "absolute_path": True,
+                    },
+                ),
+            ),
+            (
+                "out_2",
+                attr.ib(
+                    type=File,
+                    metadata={
+                        "help_string": "fictional output #1",
+                        "output_file_template": "{out_dir}/test.nii",
+                        "absolute_path": True,
+                    },
+                ),
+            ),
+        ],
+        bases=(ShellOutSpec,),
+    )
+
+    shelly = ShellCommandTask(
+        executable="touch", input_spec=input_spec, output_spec=out_spec, out_dir=tmpdir
+    )
+    shelly()
+    res = shelly.result()
+    # checking if the output was created
+    assert (Path(tmpdir) / Path("test.nii")).exists()
+    # check if path of the out_1 is correct
+    assert res.output.out_1 == attr.NOTHING
+    # check if path of the out_1 is correct
+    assert res.output.out_2 == (Path(tmpdir) / Path("test.nii"))
+
+
+def test_shellspec_cmd_absolute_path_non_existing_multi(tmpdir):
+    """test the 'absolute_path' metadata option. An output path marked with this should not be appended
+    to the nodes output_dir. Testing for a existing and non existing file in multiOutputObj."""
+    input_spec = SpecInfo(
+        name="Input",
+        fields=[
+            (
+                "out_name",
+                attr.ib(
+                    type=MultiInputObj,
+                    metadata={
+                        "help_string": """
+                        base name of the pretend outputs.
+                        """,
+                        "mandatory": True,
+                        "argstr": "...",
+                    },
+                ),
+            )
+        ],
+        bases=(ShellSpec,),
+    )
+    out_spec = SpecInfo(
+        name="Output",
+        fields=[
+            (
+                "out_list",
+                attr.ib(
+                    type=MultiOutputFile,
+                    metadata={
+                        "help_string": "fictional output #1",
+                        "output_file_template": "{out_name}",
+                        "absolute_path": True,
+                    },
+                ),
+            ),
+        ],
+        bases=(ShellOutSpec,),
+    )
+
+    shelly = ShellCommandTask(
+        executable="touch",
+        input_spec=input_spec,
+        output_spec=out_spec,
+        out_name=[str(Path(tmpdir) / "test_1.nii"), "test_2.nii"],
+    )
+    shelly()
+    res = shelly.result()
+    # checking if the outputs are Nothing
+    assert res.output.out_list[0] == Path(tmpdir) / "test_1.nii"
+    assert res.output.out_list[1] == attr.NOTHING
```

Comment:
I'm not sure if I wouldn't expect an error here

Reply:
i think it just might be very misleading, and I'm not sure if they are any good  advantages

Reply:
So, a MultiOutput should raise an error if one of the outputs is not existing, or only if it set to mandatory ?

Reply:
I'm not sure if this is about `MultiOutputPath`
perhaps I don't understand something more general. I thought that "absolute_path" should give you an option to save something outside the working directory if you provide an absolut path.
In this test, both files are created, but the second one doesn't have proper path (so it's treated as the relative path).
So perhaps we should inform the users that something is wrong with path instead of creating the file, but not sending it to the output

Reply:
With the second file I am testing that if ```absolute_path``` is specified that a file in the cache directory is not returned, i.e., I am double checking that the provided file path (here just 'test_2.nii') is not appended to the path of the nodes output directory. I am just creating the file because the name needs to be in the ```MultiInputObj``` such that it can be searched by the ```MultiOutputFile```.

Reply:
but the only reason why the second file is not appended to the output directory is because `test_2.nii'` is not the correct absolute path. Am i getting this right?

Reply:
Yes, no file exists at that location.